ecs-cli/modules/config/readwriter.go

```diff
@@ -46,42 +41,79 @@ type ReadWriter interface {
 // compose-project-name-prefix = ecscompose-
 // compose-service-name-prefix =
 // cfn-stack-name-prefix = ecs-cli-
-
-type IniReadWriter struct {
+type INIReadWriter struct {
 	*Destination
 	cfg *ini.File
 }
 
-// NewReadWriter creates a new Parser object.
-func NewReadWriter() (*IniReadWriter, error) {
-	dest, err := newDefaultDestination()
-	if err != nil {
-		return nil, err
-	}
+// NewINIReadWriter creates a new Ini Parser object for the old ini configs
+func NewINIReadWriter(dest *Destination) (*INIReadWriter, error) {
 
-	iniCfg, err := newIniConfig(dest)
+	iniCfg, err := newINIConfig(dest)
 	if err != nil {
 		return nil, err
 	}
 
-	return &IniReadWriter{Destination: dest, cfg: iniCfg}, nil
+	return &INIReadWriter{Destination: dest, cfg: iniCfg}, nil
 }
 
 // GetConfig gets the ecs-cli config object from the config file.
-func (rdwr *IniReadWriter) GetConfig() (*CliConfig, error) {
-	to := new(CliConfig)
-	err := rdwr.cfg.MapTo(to)
+// map contains the keys that are present in the config file (maps string field name to string field value)
+// map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
+// cause certain keys to be mapped to maps of keys
+func (rdwr *INIReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
+	configMap := make(map[interface{}]interface{})
+	to := &CLIConfig{}
+
+	// read old ini formatted file
+	oldFormat := &iniCLIConfig{iniSectionKeys: new(iniSectionKeys)}
+	err := rdwr.cfg.MapTo(oldFormat)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return to, nil
+	// Create the configMap
+	if rdwr.IsKeyPresent(ecsSectionKey, "cluster") {
```

Comment:
I can remove this if you want- however this code will be deleted when the next pull request happens for the rest of the multiple cluster config. 

(So the reason I did it this way was so that the behavior of the map if the file is ini is the same as if it was yaml. If the file is yaml and cluster is somehow not present, then there will be no key for it in the map. Basically the contract for the GetConfig function would state that map contains the keys present in the config file (regardless of whether or not we have an intent to use them in the application).
Obviously, either way the code works, it won't matter because if the cluster is not present then eventually an error will be thrown.)